dev/com.ibm.ws.jpa.container.core/src/com/ibm/ws/jpa/management/JPAIntrospection.java

```diff
@@ -166,92 +252,133 @@ public void executeIntrospectionAnalysis(final PrintWriter dout) {
-                dout.println("   Persistence Unit Roots:");
-                final Map<URL, List<JPAPUnitInfoIntrospect>> urlPUInfoMap = entry.getValue().getURL_JPAPuInfoMap();
-                if (urlPUInfoMap.size() == 0) {
-                    dout.println("      None");
-                } else {
-                    for (Map.Entry<URL, List<JPAPUnitInfoIntrospect>> e2 : urlPUInfoMap.entrySet()) {
-                        dout.println("      " + getShortenedURLPath(e2.getKey()) + " (" + e2.getValue().size() + " persistence units)");
-                    }
-                }
-            }
+        }
 
-            dout.println();
+        return analysisResultMap;
+    }
+
+    private void doExecuteTraceAnalysis() {
+        if (!(tc.isAnyTracingEnabled() && tc.isDebugEnabled())) {
+            return;
+        }
+
+        try {
+            final Map<String, JPAAnalysisResult> analysisResultMap = executeDiagnostics(false);
 
             for (Map.Entry<String, JPAAnalysisResult> entry : analysisResultMap.entrySet()) {
                 final JPAAnalysisResult result = entry.getValue();
-                final PersistenceUnitScannerResults2 scannerResults = result.getAppl().results;
-
-                dout.println();
-                dout.println("################################################################################");
-                dout.println("Application \"" + entry.getKey() + "\":");
+                final JPAApplInfoIntrospect applInfoIntro = result.getAppl();
+                final PersistenceUnitScannerResults results = applInfoIntro.results;
 
-                if (scannerResults == null) {
-                    dout.println("   No JPA Materials to Analyze.");
+                if (results == null) {
                     continue;
                 }
 
-                dout.println("   Total ORM Files: " + scannerResults.getAllEntityMappingsDefinitions().size());
+                final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                results.generateORMDump(new PrintWriter(baos));
 
-                {
-                    final Map<URL, Set<ClassInfoType>> urlCitMap = scannerResults.getAllScannedClasses();
-                    int count = 0;
-                    for (Set<ClassInfoType> citSet : urlCitMap.values()) {
-                        count += citSet.size();
-                    }
+                final StringBuilder sb = new StringBuilder();
+                sb.append("\nJPA Diagnostic Dump for Application " + entry.getKey() + " : \n");
+                sb.append(baos.toString());
+
+                Tr.debug(tc, "JPAORMDiagnostics Dump", sb.toString());
+            }
+        } catch (Throwable t) {
+            FFDCFilter.processException(t, JPAIntrospection.class.getName() + ".executeTraceAnalysis", "185");
+        }
+    }
+
+    private void doExecuteIntrospectionAnalysis(final PrintWriter dout) {
```

Comment:
Should methods like this have a try/catch to ensure any error we encounter during introspection does not prevent the application itself from working?  Seems that an application that could start and run just fine may fail to start if we have a bug in some of our introspection code, like this method.

Reply:
Good point, wrapping the static public method body impls with a catch block that logs to ffdc.